src/server/middleware/auth.ts

```diff
@@ -278,6 +282,11 @@ const handleKeypairAuth = async (
       throw error;
     }
 
+    if (data && req.method === "POST" && data !== hashRequestBody(req)) {
+      error = "The request body has been tampered with.";
+      throw error;
+    }
```

Comment:
We can only do this check when the request is of POST type, otherwise the req.body will come as empty

Reply:
Added the `POST` check as `GET` spec doesn't support body as per the HTTP/1.1. Some web servers (Apache, Nginx & Traefik) drop the body param for GET request when forwarding the request. The spec doesn't forbid it, but the HTTP servers do drop it.